plugins/slack_webhooks/app/decorators/stage_decorator.rb

```diff
@@ -6,6 +6,14 @@ def send_slack_webhook_notifications?
     slack_webhooks.any?
   end
 
+  def slack_buddy_channels
+    slack_webhooks.for_buddy.map(&:channel)
```

Comment:
although, if you check out section 1.7 of http://edgeguides.rubyonrails.org/caching_with_rails.html#low-level-caching and look at the timings reported in the log you posted, maybe its not essential.